lib/ansible/modules/windows/win_audit_rule.ps1

```diff
@@ -0,0 +1,182 @@
+
+    $HT = Foreach ($Obj in $ACL.Audit)
+    {
+        @{
+        identity_reference = $Obj.IdentityReference.ToString()
+        rights = ($Obj | select -expand "*rights").ToString()
+        audit_flags = $Obj.AuditFlags.ToString()
+        is_inherited = $Obj.IsInherited.ToString()
+        inheritance_flags = $Obj.IsInherited.ToString()
+        propagation_flags = $Obj.PropagationFlags.ToString()
+        }
+    }
+
+    $HT
+}
+
+$result = @{
+    changed = $false
+    # identity_reference = $identity_reference
+    # path = $path
+    # rights = $rights
+    # inheritance_flags = $inheritance_flags
+    # propagation_flags = $propagation_flags
+    # audit_flags = $audit_flags
+    # state = $state
+    current_audit_rules = Get-CurrentAuditRules $path
+}
+
+#Make sure target path is valid
+If (-not (Test-Path -Path $path) )
+{
+    Fail-Json -obj $result -message "defined path ($path) is not found/invalid"
+}
+
+#Make sure identity is valid and can be looked up
+Try {
+    $SID = Convert-ToSid $identity_reference
+}
+Catch {
+    Fail-Json -obj $result -message "Failed to lookup the identity ($identity_reference) - $($_.exception.message)"
+}
+
+#Make sure that the defined rights are valid and define type of audit rule
+If ($path -match "^HK(CC|CR|CU|LM|U):\\" -and $state -eq 'present')
+{
+    $PossibleRights = [System.Enum]::GetNames([System.Security.AccessControl.RegistryRights])
+    ($rights -split ',').Trim() | % {
+        if ($_ -notin $PossibleRights)
+        {
+            Fail-Json -obj $result -message "$_ does not seem to be a valid REGISTRY right"
+        }
+    }
+    $Registry = $True
+}
+Elseif ($state -eq 'present')
+{
+    $PossibleRights = [System.Enum]::GetNames([System.Security.AccessControl.FileSystemRights])
+    ($rights -split ',').Trim() | % {
+        if ($_ -notin $PossibleRights)
+        {
+            Fail-Json -obj $result -message "$_ does not seem to be a valid FILE SYSTEM right"
+        }
+    }
+    $FileSystem = $True
+}
+
+$ACL = Get-Acl $path -Audit
+
+#configure acl object to remove the specified user
+If ($state -eq 'absent')
+{
+    #Try and find an identity on the object that matches identity_reference
+    $ToRemove = ($ACL.Audit | ? {$_.IdentityReference.Translate([System.Security.Principal.SecurityIdentifier]) -eq $SID}).IdentityReference
+
+    #Exit with changed false if no identity is found
+    If (-Not $ToRemove)
+    {
+        $result.current_audit_rules = Get-CurrentAuditRules $path
+        Exit-Json -obj $result -message
+    }
+
+    #update the ACL object if identity found
+    Try
+    {
+        $ACL.PurgeAuditRules($ToRemove)
+    }
+    Catch
+    {
+        $result.current_audit_rules = Get-CurrentAuditRules $path
+        fail-Json -obj $result -message "Failed to remove auditrule: $($_.Exception.Message)"
+    }
+
+}
+#otherwise configure acl object to add/modify user
+Else
+{
+    $flags = [System.Security.AccessControl.AuditFlags]$audit_flags
```

Comment:
Does this work if you have multiple entries?

Reply:
it seems to which is great. makes changing the type to list a lot simpler.